datadog/resource_datadog_dashboard.go

```diff
@@ -4593,6 +4843,162 @@ func getTimeseriesRequestSchema() map[string]*schema.Schema {
 		},
 	}
 }
+
+func buildDatadogFormula(data map[string]interface{}) datadogV1.WidgetFormula {
+	formula := datadogV1.WidgetFormula{}
+	if formulaExpression, ok := data["formula_expression"].(string); ok && len(formulaExpression) != 0 {
+		formula.Formula = formulaExpression
+	}
+	if alias, ok := data["alias"].(string); ok && len(alias) != 0 {
+		formula.SetAlias(alias)
+	}
+	if limits, ok := data["limit"].([]interface{}); ok && len(limits) != 0 {
+		datadogLimit := datadogV1.NewWidgetFormulaLimit()
+		limit := limits[0].(map[string]interface{})
+		if count, ok := limit["count"].(int); ok && count != 0 {
+			datadogLimit.SetCount(int64(count))
+		}
+		if order, ok := limit["order"].(string); ok && len(order) > 0 {
+			datadogLimit.SetOrder(datadogV1.QuerySortOrder(order))
+		}
+		formula.SetLimit(*datadogLimit)
+	}
+	return formula
+}
+
+func buildDatadogEventQuery(data map[string]interface{}) datadogV1.FormulaAndFunctionQueryDefinition {
+	dataSource := datadogV1.FormulaAndFunctionEventsDataSource(data["data_source"].(string))
+	computeList := data["compute"].([]interface{})
+	computeMap := computeList[0].(map[string]interface{})
+	aggregation := datadogV1.FormulaAndFunctionEventAggregation(computeMap["aggregation"].(string))
+	compute := datadogV1.NewTimeSeriesFormulaAndFunctionEventQueryDefinitionCompute(aggregation)
+	if interval, ok := computeMap["interval"]; ok && interval.(int) != 0 {
+		compute.SetInterval(int64(interval.(int)))
+	}
+	if metric, ok := computeMap["metric"]; ok && len(metric.(string)) > 0 {
+		compute.SetMetric(metric.(string))
+	}
+	eventQuery := datadogV1.NewTimeSeriesFormulaAndFunctionEventQueryDefinition(*compute, dataSource)
+	if v, ok := data["name"].(string); ok && len(v) != 0 {
+		eventQuery.Name = &v
```

Comment:
```suggestion
		eventQuery.SetName(v)
```
Lets use the setters here

Reply:
Done 👍